### PR TITLE
Admin app tickets

### PR DESCRIPTION
**Pull request by ArinaSavchenko**

### CODE REVIEW

File: AdminApp/src/app/Services/ticket-type.service.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  getTicketTypes(): Observable<TicketType[]> {
+    return this.http.get<TicketType[]>( this.ticketTypesUrl );
+  }
+
+  getTicketType(id: number): Observable<TicketType> {
+    if (id === null) {
+      return null;
+    }
+    const url = `${this.ticketTypesUrl}/${id}`;
+    return this.http.get<TicketType>( url );
```
Comment (andrei-kruhliakou-itechart) [inline AdminApp/src/app/Services/ticket-type.service.ts]: It is better not to use spaces around method arguments, e.g.
return this.http.get<TicketType>(url);

File: Airline_Web_API/Airline_Web_API/Controllers/FlightsController.cs [hunk 2]
```diff
@@ -65,9 +65,9 @@ public async Task<ActionResult<FlightViewModel>> GetFlightParallel([FromQuery] i
         [HttpPost]
         public async Task<ActionResult> PostFlight([FromBody] NewFlightModel model)
         {
-            await _flightService.AddFlightAsync(model);
+            var fligthId = await _flightService.AddFlightAsync(model);
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Controllers/FlightsController.cs]: typo in 'flightId'

File: Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TicketTypeViewModel>>> GetTicketTypes()
+        {
+            var ticketTypes = await _ticketTypeService.GetTicketTypesAsync();
+
+            if (ticketTypes == null)
+            {
+                return NoContent();
+            }
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs]: More common alternative for endpoints that return collection will be just returning empty collection.
That way you don't need ActionResult.
BTW, is it possible to get null in ticketTypes?

File: Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TicketTypeViewModel>> GetTicketType(int id)
+        {
+            var ticketType = await _ticketTypeService.GetTicketTypeByIdAsync(id);
+
+            if (ticketType == null)
+            {
+                return NoContent();
+            }
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs]: If ticket type with specified id doesn't exist shouldn't it be NotFound?

File: Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            return Ok(ticketType);
+        }
+
+        [Authorize(Roles = "admin")]
+        [HttpPost]
+        public async Task<ActionResult> Post([FromBody] TicketTypeViewModel model)
+        {
+            await _ticketTypeService.AddTicketTypeAsync(model);
+
+            return Ok();
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs:51]: Alternative would be CreatedAt but Ok is also fine.

File: Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        [Authorize(Roles = "admin")]
+        [HttpPut]
+        public async Task<ActionResult<Response<string>>> Put([FromBody] TicketTypeViewModel model)
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs]: If you're trying to follow best practices for RESTful API shouldn't put request have a bit different url? Is there something missing here?

File: Airline_Web_API/Airline_Web_API/Models/TicketType.cs [hunk 7]
```diff
@@ -1,22 +1,25 @@
...
-        public int? BaggageNumber { get; set; }
-        public int? BaggageMaxWeight { get; set; }
+        public int CarryOnBagsNumber { get; set; }
+        public int CarryOnBagMaxWeight { get; set; }
+        public int BaggageNumber { get; set; }
+        public int BaggageMaxWeight { get; set; }
+        public decimal PricePerExtraCarryOnBag { get; set; }
+	    public decimal PricePerExtraCarryOnBagKg { get; set; }
+	    public decimal PricePerExtraBaggage { get; set; }
+	    public decimal PricePerExtraBaggageKg { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Models/TicketType.cs]: Indent

File: Airline_Web_API/Airline_Web_API/Services/TicketTypeService.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        private readonly AirlineContext _context;
+        private readonly IMapper _mapper;
+
+        public TicketTypeService(AirlineContext airlineContext, IMapper mapper)
+        {
+            _context = airlineContext;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<TicketTypeViewModel>> GetTicketTypesAsync()
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Services/TicketTypeService.cs:24]: Why this method returns IEnumerable but the same method in TicketsService returns List? Which is correct?

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs (hunk 3 above)
Comment (ArinaSavchenko) [reply Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs]: Oh, this chick is unnecessary here. I've corrected it

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

**Pull request merged**